src/contracts/fungible/cache/file.rs

```diff
@@ -192,4 +192,34 @@ impl Cache for FileCache {
         self.save()?;
         Ok(existed)
     }
+
+    fn utxo_allocation_map(
+        &self,
+        contract_id: ContractId,
+    ) -> Result<BTreeMap<bitcoin::OutPoint, AtomicValue>, CacheError> {
+        let asset = self.asset(contract_id).unwrap();
+
+        let allocation_map = asset.known_allocations();
+
+        let mut result = BTreeMap::new();
+
+        for item in allocation_map {
+            let mut sum = 0;
+            for alloc in item.1 {
+                sum += alloc.value().value;
+            }
+
+            result.insert(item.0.clone(), sum);
+        }
+
+        Ok(result)
+    }
+
+    // TODO: Implement it for FileCache
+    fn asset_allocation_map(
+        &self,
+        _utxo: &bitcoin::OutPoint,
+    ) -> Result<BTreeMap<String, u32>, CacheError> {
+        unimplemented!();
```

Comment:
This is quite simple: we have all data in memory in form of a data structure. We just need to access all assets data with `Cache::assets()` method and iterate over them collecting all items corresponding to the given outpoint.